contracts/contracts/sbtc-withdrawal.clar

```diff
@@ -33,6 +45,165 @@
+                                     bitcoin-txid: (optional (buff 32)), 
+                                     output-index: (optional uint), 
+                                     fee: (optional uint)})))
+  (let 
+      (
+          (current-signer-data (contract-call? .sbtc-registry get-current-signer-data))
+      )
+
+      ;; Check that the caller is the current signer principal
+      (asserts! (is-eq (get current-signer-principal current-signer-data) tx-sender) ERR_INVALID_CALLER)
+
+      (try! (match (get agg-errs (unwrap-panic (fold complete-individual-withdrawal-helper withdrawals (ok {index: u0, agg-errs: none}))))
+        ;; agg-errs exist
+        agg-errs
+          (err 
+            (unwrap-panic (string-to-uint? (concat ERR_WITHDRAWALS_PREFIX agg-errs))))
+        ;; agg-errs free (all good)
+        (ok true)
+      ))
+
+      (ok true)
+  )
+)
+
+;; Loop through each submission
+;;  Regardless of whether there are errors or not we need to try every submission
+;;  If it's okay, then we just increase the index by one
+;;  If it's not okay, we need create and/or store an error message
+;; At the end, if the error message is empty, all were okay / we return okay
+(define-private (complete-individual-withdrawal-helper (withdrawal 
+                                                        {request-id: uint, 
+                                                        status: bool, 
+                                                        signer-bitmap: uint, 
+                                                        bitcoin-txid: (optional (buff 32)), 
+                                                        output-index: (optional uint), 
+                                                        fee: (optional uint)}) 
+                                                       (helper-response (response {index: uint, agg-errs: (optional (string-ascii 100))} uint)))
+
+  (let
+    (
+      (current-request-id (get request-id withdrawal))
+      (current-signer-bitmap (get signer-bitmap withdrawal))
+      (current-bitcoin-txid (get bitcoin-txid withdrawal))
+      (current-output-index (get output-index withdrawal))
+      (current-fee (get fee withdrawal))
+      (current-index (get index (unwrap-panic helper-response)))
+      (current-agg-errs (get agg-errs (unwrap-panic helper-response)))
+    )
+    ;; check / unwrap existing errors
+    (match current-agg-errs 
+      existing-errs
+        (if (get status withdrawal)
+          ;; accepted
+          (match (accept-withdrawal-request (get request-id withdrawal) (unwrap-panic current-bitcoin-txid) current-signer-bitmap (unwrap-panic current-output-index) (unwrap-panic current-fee))
+            ok-resp
+              (ok {index: (+ u1 current-index), agg-errs: current-agg-errs})
+            err-resp
+              (ok {index: (+ u1 current-index), agg-errs: (as-max-len? (concat existing-errs (int-to-ascii current-index)) u100)})
```

Comment:
I think the `agg-errs` part needs to change. The issue is that the `int-to-ascii` does not return a string with a fixed length. So the issue is that we cannot differentiate errors for requests with indices 1 and 2 from the case where you get one error at index 12.

Also, the max length should be 19 in order to stay within the max of a u128 (assuming each string entry is two characters).

Reply:
@djordon I reverted it all. Should be ready for approval.